app/generic/proof/facades/acapy_proof_v1.py

```diff
@@ -0,0 +1,180 @@
+            if free:
+                presentation_exchange = (
+                    await controller.present_proof_v1_0.send_request_free(
+                        body=V10PresentationSendRequestRequest(**proof_request.dict())
+                    )
+                )
+            elif isinstance(proof_request, AdminAPIMessageTracing) and pres_ex_id:
+                presentation_exchange = (
+                    await controller.present_proof_v1_0.send_request(
+                        pres_ex_id=pres_ex_id, body=proof_request
+                    )
+                )
+            elif isinstance(proof_request, V10PresentationProposalRequest):
+                presentation_exchange = (
+                    await controller.present_proof_v1_0.send_proposal(
+                        body=proof_request
+                    )
+                )
+            else:
+                raise NotImplementedError
+            return cls.__record_to_model(presentation_exchange)
+        except Exception as e:
+            raise e from e
+
+    @classmethod
+    async def accept_proof_request(
+        cls,
+        controller: AcaPyClient,
+        pres_ex_id: str,
+        body: IndyPresSpec,
+    ) -> PresentationExchange:
+        pres_ex_id = cls.__pres_id_no_version(pres_ex_id=pres_ex_id)
+        presentation_record = await controller.present_proof_v1_0.send_presentation(
+            pres_ex_id=pres_ex_id, body=body
+        )
+        return cls.__record_to_model(presentation_record)
+
+    @classmethod
+    async def reject_proof_request(
+        cls,
+        controller: AcaPyClient,
+        pres_ex_id: str,
+        problem_report: str = None,
+    ) -> None:
+        # get the record
+        pres_ex_id = cls.__pres_id_no_version(pres_ex_id=pres_ex_id)
+        proof_request = await controller.present_proof_v1_0.get_record(
+            pres_ex_id=pres_ex_id
+        )
+        # Report problem if desired
+        if problem_report:
+            try:
+                await controller.present_proof_v1_0.report_problem(
+                    pres_ex_id=pres_ex_id,
+                    body=V10PresentationProblemReportRequest(
+                        description=problem_report
+                    ),
+                )
+            except Exception as e:
+                raise e from e
+        # delete exchange record
+        delete_proof_request_res = await controller.present_proof_v1_0.delete_record(
+            pres_ex_id=pres_ex_id
+        )
+        if not isinstance(proof_request, V10PresentationExchange) or not isinstance(
+            delete_proof_request_res, (Dict, NoneType)
+        ):
+            raise HTTPException(status_code=500, detail="Failed to delete record")
+
+    @classmethod
+    def __pres_id_no_version(cls, pres_ex_id: str) -> str:
+        if pres_ex_id.startswith("v1-") or pres_ex_id.startswith("v2-"):
+            return pres_ex_id[3:]
+        else:
+            return pres_ex_id
```

Comment:
I think such convenience methods (support ids with and without `v1-` prefix) may make the API easier in the start, but I think it is good to have a **right** and a **wrong** way to call an API. So I would say we only allow one of the two